src/plugins/share/common/config.ts

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { schema, TypeOf } from '@kbn/config-schema';
+
+export const configSchema = schema.object({
+  new_version: schema.object({
+    enabled: schema.boolean({
+      defaultValue: false,
+      validate(value) {
+        if (value !== true && value !== false) {
+          return `cannot use values beyond true or false`;
+        }
```

Comment:
__question:__ wouldn't `@kbn/config-schema` reject non-boolean values anyway without custom validator function?

Reply:
Trying it out! Removed in [dfab294](https://github.com/elastic/kibana/pull/178035/commits/dfab294ddc4c745aee82862630545eb707f5e952)